components/kyma-environment-broker/internal/model.go

```diff
@@ -529,19 +528,23 @@ func NewSuspensionOperationWithID(operationID string, instance *Instance) Deprov
 			UpdatedAt:       time.Now(),
 			Type:            OperationTypeDeprovision,
 			InstanceDetails: instance.InstanceDetails,
-			FinishedStages:  make(map[string]struct{}, 0),
+			FinishedStages:  make([]string, 0),
 			Temporary:       true,
 		},
 	}
 }
 
 func (o *Operation) FinishStage(stageName string) {
-	o.FinishedStages[stageName] = struct{}{}
+	o.FinishedStages = append(o.FinishedStages, stageName)
```

Comment:
curious if we want to check first if some previous execution of operation processing already made the stage finished to avoid duplicates in the filed?